src/nwb_datajoint/spikesorting/spikesorting_curation.py

```diff
@@ -381,17 +401,18 @@ def _compute_metric(self, waveform_extractor, metric_name, **metric_params):
         metric_func = _metric_name_to_func[metric_name]
         # TODO clean up code below
         if metric_name == 'isi_violation':
-            metric = metric_func([], waveform_extractor, **metric_params)
-        elif metric_name == 'snr':
+            metric = metric_func(waveform_extractor, **metric_params)
+        elif (metric_name == 'snr' or
+              metric_name == 'peak_offset'):
             peak_sign = metric_params['peak_sign']
             del metric_params['peak_sign']
             metric = metric_func(waveform_extractor,
                                  peak_sign=peak_sign, **metric_params)
         else:
-            metric = {}
-            for unit_id in waveform_extractor.sorting.get_unit_ids():
-                metric[str(unit_id)] = metric_func(
-                    waveform_extractor, this_unit_id=unit_id, **metric_params)
+            metric = {str(unit_id): metric_func(waveform_extractor,
```

Comment:
Note that here I made `unit_id` a string because they have to be string in order for them to be exported to `sortingview`. but it is not clear to me what we need to make them string here - maybe we just need to check and make the change when actually adding the metrics to a workspace. if you agree, let's get rid of the str()

Reply:
Wonder if we should change `sortingview` so that's not a requirement...